llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -635,25 +635,23 @@ static bool usedInGlobalVarDef(const Constant *C) {
   return false;
 }
 
-static bool usedInOneFunc(const User *U, Function const *&oneFunc) {
-  if (const GlobalVariable *othergv = dyn_cast<GlobalVariable>(U)) {
-    if (othergv->getName() == "llvm.used")
+static bool usedInOneFunc(const User *U, Function const *&OneFunc) {
```

Comment:
`return bool + value via argument` may be converted to returning std::optional.

Definitely easier to grok than `const *& something`.

Reply:
I gave this a go but unfortunately replicating the logic with std::optional ends up being a bit cumbersome and slightly longer than the current version. I'm leaving this unchanged for now.